containers/daap-delete-data-product/tests/unit/conftest.py

```diff
@@ -205,3 +205,51 @@ def create_failed_raw_and_curated_data(
                 Key=f"fail/{data_product_name}/{version}/schema0/fail-file-{str(i)}.json",
                 Body=json.dumps({"content": f"{i}"}),
             )
+
+
+@pytest.fixture
+def create_curated_data(
+    s3_client,
+    create_curated_bucket,
+    data_product_name,
+    data_product_versions,
+):
+    for version in data_product_versions:
+        for i in range(10):
+            s3_client.put_object(
+                Bucket=os.getenv("CURATED_DATA_BUCKET"),
+                Key=f"curated/{data_product_name}/{version}/schema0/curated-file-{str(i)}.json",
+                Body=json.dumps({"content": f"{i}"}),
+            )
```

Comment:
DRY suggestion:
Pull out this loop to a `populate_bucket` function that takes `bucket_name` and `bucket_type` as additional args then call that function for these three `create_x_data` functions

(This could also be shrunk into a single fixture `populate_buckets` that loops over the three sets of bucket args)

Reply:
`@pytest.fixture
def create_fail_data(
    s3_client,
    create_raw_bucket,
    data_product_name,
    data_product_versions,
):
    populate_buckets(
        data_product_versions,
        s3_client,
        data_product_name,
        "RAW_DATA_BUCKET",
        "fail",
    )`